lib/src/Uncertainty/Algorithm/Simulation/LineSampling.cxx

```diff
@@ -0,0 +1,282 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Line sampling simulation method
+ *
+ *  Copyright 2005-2025 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/LineSampling.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/DistFunc.hxx"
+#include "openturns/ComposedFunction.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/LinearFunction.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(LineSampling)
+
+static const Factory<LineSampling> Factory_LineSampling;
+
+
+/* Default constructor */
+LineSampling::LineSampling()
+  : EventSimulation()
+{
+}
+
+
+/* Constructor with parameters */
+LineSampling::LineSampling(const RandomVector & event,
+                           const Point & initialAlpha,
+                           const RootStrategy & rootStrategy)
+  : EventSimulation(event.getImplementation()->asComposedEvent())
+  , initialAlpha_(initialAlpha)
+  , rootStrategy_(rootStrategy)
+  , searchOppositeDirection_(ResourceMap::GetAsBool("LineSampling-DefaultSearchOppositeDirection"))
+  , adaptiveImportantDirection_(ResourceMap::GetAsBool("LineSampling-DefaultAdaptiveImportantDirection"))
+{
+  if (!event.isEvent() || !event.isComposite()) throw InvalidArgumentException(HERE) << "LineSampling requires a composite event";
```

Comment:
```suggestion
  if (!(event.isEvent() && event.isComposite())) throw InvalidArgumentException(HERE) << "LineSampling requires a composite event";
```
Clearer IMO, because the sufficient condition for the class to work is "the RandomVector is an event and is composite" and should be clearly expressed.